administrator/components/com_associations/View/Autoassoc/HtmlView.php

```diff
@@ -0,0 +1,178 @@
+<?php
+/**
+ * @package     Joomla.Administrator
+ * @subpackage  com_associations
+ *
+ * @copyright   Copyright (C) 2005 - 2018 Open Source Matters, Inc. All rights reserved.
+ * @license     GNU General Public License version 2 or later; see LICENSE.txt
+ */
+namespace Joomla\Component\Associations\Administrator\View\Autoassoc;
+
+defined('_JEXEC') or die;
+
+use Joomla\CMS\MVC\View\HtmlView as BaseHtmlView;
+use Joomla\CMS\Language\Associations;
+use Joomla\Component\Associations\Administrator\Helper\AssociationsHelper;
+use Joomla\Registry\Registry;
+
+/**
+ * View class for automatic associations
+ *
+ * @since  __DEPLOY_VERSION__
+ */
+class HtmlView extends BaseHtmlView
+{
+	/**
+	 * An array of items
+	 *
+	 * @var   array
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $items;
+
+	/**
+	 * The pagination object
+	 *
+	 * @var    \Joomla\CMS\Pagination\Pagination
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $pagination;
+
+	/**
+	 * The model state
+	 *
+	 * @var    object
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	protected $state;
+
+	/**
+	 * Selected item type properties.
+	 *
+	 * @var    Registry
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public $itemType = null;
+
+	/**
+	 * Display the view
+	 *
+	 * @param   string  $tpl  The name of the template file to parse; automatically searches through the template paths.
+	 *
+	 * @return  void
+	 *
+	 * @since  __DEPLOY_VERSION__
+	 */
+	public function display($tpl = null)
+	{
+		$this->state         = $this->get('State');
+		$this->filterForm    = $this->get('FilterForm');
+		$this->activeFilters = $this->get('ActiveFilters');
+
+		if (!Associations::isEnabled())
```

Comment:
So many uncommented lines, don't forget your code will be read by other people and it's hard to modify code that you don't know what is doing.

I suggest you do the following:

```
// Comment
<block of code>
```

For example:

```
// Getting form data
$name = $input->get('name');
$age  = $input->get('age');

// Sanitizing name
if (strlen($name) === 0)
{
    \JFactory::getApplication()->enqueueMessage(\JText::sprintf('ERROR_MSG_NAME', $link), 'error');
}

// Sanitizing age
if ($age < MIN_AGE || $age > MAX_AGE)
{
     \JFactory::getApplication()->enqueueMessage(\JText::sprintf('ERROR_MSG_AGE', $link), 'error');
}

// Preparing an array to store into the DB
$data['name'] = $name;
$data['age']  = $age;

// Store form data into the db
$model->store($data);
```

This was just an example, not real code, so you can make better comments 😉